src/main/java/ui/FXMLLoader.java

```diff
@@ -0,0 +1,23 @@
+package ui;
+
+import org.springframework.context.ApplicationContext;
+import org.springframework.context.ApplicationContextAware;
+
+/**
+ * Define a loader for FXML that is aware of the current Spring application
+ * context. This loader uses {@link ApplicationContext#getBean(Class)} to
+ * load its controllers, and the Beans that control the UI have been
+ * initialized in the Spring IOC container
+ */
+public class FXMLLoader extends javafx.fxml.FXMLLoader implements
```

Comment:
Define an interface for this loader, and rename this class to ``ContextAwareFXMLLoader``. Mention that some encapsulation is going to be needed if we want a context-unaware FXMLLoader.